src/controllers/adminController.ts

```diff
@@ -22,13 +22,30 @@ export const getUsers = async (req: Request, res: Response) => {
 export const updateUserRole = async (req: Request, res: Response) => {
   try {
     const { userId, newRoles } = req.body;
-    if (!userId || !newRoles) {
+
+    if (!userId || !newRoles || !Array.isArray(newRoles)) {
       return res.status(400).json({
         success: false,
         message: "Invalid input",
       } as ApiResponse<null>);
     }
+
+    if (!req.user || !req.user.role.includes("admin")) {
```

Comment:
This will not satisfy what you are trying to achieve if user exist. 

In general, you should validate the user's role based on the token rather than the request payload. This is because a malicious user could manipulate the request and pass a different role to gain unauthorized access or modify data.